### PR TITLE
[opt](nereids)improve partition prune when Date function is used

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
 F: a DateTime or DateTimeV2 column
 Date(F) > 2020-01-01 => F > 2020-01-02 00:00:00
 Date(F) >= 2020-01-01 => F > 2020-01-01 00:00:00
 Date(F) in (2020-01-02, 2020-01-01) =>
               (2020-01-01 24:00:00 >= F >= 2020-01-01 00:00:00)
               or (2020-01-02 24:00:00 >= F >= 2020-01-02 00:00:00)
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/PredicateRewriteForPartitionPrune.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+/**
+ * F: a DateTime or DateTimeV2 column
+ * Date(F) > 2020-01-01 => F > 2020-01-02 00:00:00
+ * Date(F) >= 2020-01-01 => F > 2020-01-01 00:00:00
+ * Date(F) in (2020-01-02, 2020-01-01) =>
+ *              (2020-01-01 24:00:00 >= F >= 2020-01-01 00:00:00)
+ *              or (2020-01-02 24:00:00 >= F >= 2020-01-02 00:00:00)
+ *
+ */
+public class PredicateRewriteForPartitionPrune
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/PredicateRewriteForPartitionPrune.java:52]: the in-predicate is not good for rewrite

**Comment by englefly** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**